File: java/two-fer/src/main/java/Twofer.java
public class Twofer {

  public String twofer(String name) {
    if (noNameGiven(name)) {
      return "One for you, one for me.";
    }
    return String.format("One for %s, one for me.", name);
  }

  public boolean noNameGiven(String name) {
    return name == null;
  }
}

File: go/two-fer/two_fer.go
package twofer

import "fmt"

// ShareWith returns a string in the following format: "One for you, one for me."
// if a name is provided, it is used instead of "you".
func ShareWith(name string) string {
	if name == "" {
		name = "you"
	}
	return fmt.Sprintf("One for %s, one for me.", name)
}
